packages/hdwallet-xdefi/src/xdefi.ts

```diff
@@ -220,6 +237,21 @@ export class XDEFIHDWallet implements core.HDWallet, core.ETHWallet {
     return chainId === 1;
   }
 
+  public async ethSwitchChain(chainId = 1): Promise<void> {
+    const hexChainId = ethers.utils.hexValue(chainId);
+    const provider: any = await detectEthereumProvider({ mustBeMetaMask: true, silent: false, timeout: 3000 });
+    try {
+      // at this point, we know that we're in the context of a valid MetaMask provider
+      await provider.request({ method: "wallet_switchEthereumChain", params: [{ chainId: hexChainId }] });
+    } catch (e: any) {
+      const error: core.SerializedEthereumRpcError = e;
+      if (error.code === 4902) {
+        // TODO: EVM Chains Milestone
+        // We will need to pass chainName and rpcUrls, which we don't have yet, to add a chain to XDEFI.
+      }
+    }
+  }
+
```

Comment:
Log the error for both the 4902-specific and all other cases to ease debugging.

Reply:
Added a console.error in de254daa059182713ac2e6c2d17869ff1fc55840